helm/vitess/templates/_vttablet.tpl

```diff
@@ -140,6 +142,87 @@ spec:
       shard: {{ $shardClean | quote }}
       type: {{ $tablet.type | quote }}
 
+{{ if eq $tablet.type "replica" }}
+---
+###################################
+# vttablet InitShardMaster Job
+###################################
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ $shardName }}-init-shard-master
+spec:
+  backoffLimit: 0
+  template:
+    spec:
+      restartPolicy: OnFailure
+      containers:
+      - name: init-shard-master
+        image: "vitess/vtctlclient:{{$vitessTag}}"
+
+        command: ["bash"]
+        args:
+          - "-c"
+          - |
+            set -ex
+
+            SECONDS=0
+            TIMEOUT_SECONDS=600
+
+            # poll every 5 seconds to see if vtctld is ready
+            until vtctlclient -server vtctld.{{ $namespace }}:15999 ListAllTablets {{ $cellClean }} > /dev/null 2>&1; do 
+              if (( $SECONDS > $TIMEOUT_SECONDS )); then
+                echo "timed out waiting for vtctlclient to be ready"
+                exit 1
+              fi
+              sleep 5
+            done
+
+            until [ $TABLETS_READY ]; do
+              # get all the tablets in the current cell
+              cellTablets="$(vtctlclient -server vtctld.{{ $namespace }}:15999 ListAllTablets {{ $cellClean }})"
+              
+              # filter to only the tablets in our current shard
+              shardTablets=$( echo "$cellTablets" | awk 'substr( $5,1,{{ len $shardName }} ) == "{{ $shardName }}" {print $0}')
+
+              # check for a master tablet
+              masterTablet=$( echo "$shardTablets" | awk '$4 == "master" {print $1}')
```

Comment:
It's possible that there could temporarily be no tablet of type master, even post-ISM. We need to be careful not to run ISM again in this case (especially with `-force`), since it can cause data loss.

I think a somewhat more reliable signal would be an empty `master_alias` entry in the result of `GetShard`. I think even if we transiently don't have a running master, the shard record should still contain the alias of the last known master. I'm not 100% on that though, so we should still try to think if there's a better way to be sure the shard has never been initialized.

Reply:
I can do whatever you think is the most reliable. Should I implement your `GetShard` suggestion or are you looking for a better way?